Entity/NewsItem.php

```diff
@@ -70,12 +117,111 @@ public function getContent()
     }
 
     /**
-     * Get id
+     * Set mediaDisplayOption
      *
-     * @return integer
+     * @param string $mediaDisplayOption
+     *
+     * @return NewsItem
      */
-    public function getId()
+    public function setMediaDisplayOption($mediaDisplayOption)
     {
-        return $this->id;
+        $this->mediaDisplayOption = $mediaDisplayOption;
+
+        return $this;
+    }
+
+    /**
+     * Get mediaDisplayOption
+     *
+     * @return string
+     */
+    public function getMediaDisplayOption()
+    {
+        return $this->mediaDisplayOption;
+    }
+
+    /**
+     * Add media
+     *
+     * @param Media $media
+     *
+     * @return NewsItem
+     */
+    public function addMedia(Media $media)
+    {
+        $this->medias[] = $media;
+
+        return $this;
+    }
+
+    /**
+     * Remove media
+     *
+     * @param Media $media
+     */
+    public function removeMedia(Media $media)
+    {
+        $this->medias->removeElement($media);
+    }
+
+    /**
+     * Get medias
+     *
+     * @return Collection
+     */
+    public function getMedias()
+    {
+        return $this->medias;
+    }
+
+    /**
+     * Set medias
+     *
+     * @param array $medias
+     */
+    public function setMedias(array $medias)
+    {
+        $this->medias = new ArrayCollection($medias);
+    }
+
+    /**
+     * Returns media property formatted to be used in the UI.
+     *
+     * @VirtualProperty
+     */
+    public function getMedia()
```

Comment:
this function format the medias to be used by the media-selection in the UI. the `VirtualProperty` annotation will be used to serialize the result of this function.